### PR TITLE
Mutual TLS implementation, example and tests

### PR DESCRIPTION
**Pull request by Verdent**
Signed-off-by: David Kral <david.k.kral@oracle.com>

### CODE REVIEW

File: webserver/webserver/src/main/java/io/helidon/webserver/SocketConfiguration.java [hunk 1]
```diff
@@ -118,6 +120,13 @@ default String name() {
      */
     Set<String> enabledSslProtocols();
 
+    /**
+     * Whether to require client authentication or not.
+     *
+     * @return client authentication
+     */
+    ClientAuth clientAuth();
```
Comment (tomas-langer) [inline webserver/webserver/src/main/java/io/helidon/webserver/SocketConfiguration.java:128]: Netty APIs should not be part of Helidon APIs

File: examples/webserver/mutual-tls/src/test/resources/application-test.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+server:
+  port: -1
```
[the remark this answers is not in the document]
Comment (Verdent) [reply examples/webserver/mutual-tls/src/test/resources/application-test.yaml:18]: No -1 works also :-)

[the remark this answers is not in the document]
Comment (Verdent) [reply examples/webserver/mutual-tls/src/test/resources/application-test.yaml:18]: changed

File: webserver/webserver/src/main/java/io/helidon/webserver/SocketConfiguration.java (hunk 1 above)
Comment (Verdent) [reply webserver/webserver/src/main/java/io/helidon/webserver/SocketConfiguration.java]: updated

**Pull request merged**